### PR TITLE
Update references to main branch

### PR DESCRIPTION
**Pull request by findepi**
Some docs were referring to `master`, update them to refer to `main`.

### CODE REVIEW

File: site/docs/concepts/catalog.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ There are more catalog types in addition to the ones listed here as well as cust
...
+The REST catalog was introduced in the Iceberg 0.14.0 release and provides greater control over how Iceberg catalogs are implemented. Instead of using technology-specific logic contained in the catalog clients, the implementation details of a REST catalog lives on the catalog server. If you’re familiar with Hive, this is somewhat similar to the Hive thrift service that allows access to a hive server over a single port. The server-side logic can be written in any language and use any custom technology, as long as the API follows the [Iceberg REST Open API specification](https://github.com/apache/iceberg/blob/main/open-api/rest-catalog-open-api.yaml).
```
Comment (findepi) [inline site/docs/concepts/catalog.md:46]: This old link works, but is redirecting   to the new link.

File: site/docs/how-to-release.md [hunk 2]
```diff
@@ -84,7 +84,7 @@ This step can be useful to gather ongoing patches that the community thinks shou
 
 The communication can be started via a [DISCUSS] mail on the dev@ channel and the desired tickets can be added to the github milestone of the next release.
 
-Note, creating a milestone in github requires a committer. However, a non-committer can assign tasks to a milestone if added to the list of collaborators in [.asf.yaml](https://github.com/apache/iceberg/blob/master/.asf.yaml)
+Note, creating a milestone in github requires a committer. However, a non-committer can assign tasks to a milestone if added to the list of collaborators in [.asf.yaml](https://github.com/apache/iceberg/blob/main/.asf.yaml)
```
Comment (findepi) [inline site/docs/how-to-release.md:87]: This old link works, but is redirecting   to the new link.

**Review by Fokko — APPROVED** [review]
Looks good, thanks @findepi!

**Comment by findepi** [comment]
thank you @Fokko for quick review and merge!

**Pull request merged**